speechbrain/k2_integration/train_ngram.py

```diff
@@ -0,0 +1,601 @@
+#!/usr/bin/env python3
+"""
+This is an implementation of computing Kneser-Ney smoothed language model
+in the same way as srilm. This is a back-off, unmodified version of
+Kneser-Ney smoothing, which produces the same results as the following
+command (as an example) of srilm:
+
+$ ngram-count -order 4 -kn-modify-counts-at-end -ukndiscount -gt1min 0 -gt2min 0 \
+    -gt3min 0 -gt4min 0 -text corpus.txt -lm lm.arpa
+
+The data structure is based on: kaldi/egs/wsj/s5/utils/lang/make_phone_lm.py
+The smoothing algorithm is based on: http://www.speech.sri.com/projects/srilm/manpages/ngram-discount.7.html
+
+The code below is adjusted from icefall to work with SpeechBrain.
+
+Modified by:
+    * Georgios Karakasidis 2023
+"""
+
+import io
+import math
+import os
+import re
+import sys
+import logging
+from collections import Counter, defaultdict
+from typing import Iterable, List
```

Comment:
I'm wondering, but this file does not rely on K2. I think it would be more appropriate to move this file inside of `speechbrain/lm`. However, please ensure that you are not redefining any existing methods (see: arpa.py, etc). @geoph9

Reply:
Additionally, it would be great to have some unit tests and a lot of documentation to explain how to use it.